pkg/session/session.go

```diff
@@ -2274,6 +2278,17 @@ func runStmt(ctx context.Context, se *session, s sqlexec.Statement) (rs sqlexec.
 	}
 
 	rs, err = s.Exec(ctx)
+
+	// Pipelined-DML can return assertion error and write conflict here, handle them like we handle
+	// errors returned from commit.
+	if err != nil {
```

Comment:
Should there be a condition check like
```
if s.isPipelinedDML {
    if err != nil {
        err = se.handleAssertionFailure(ctx, err)
    }
} else {
    unrechable or errors.
}
```

Reply:
Could be. It also makes the intention clearer.